cmake/multi_image.cmake

```diff
@@ -74,8 +74,25 @@ function(add_child_image_from_source)
       message(FATAL_ERROR
         "Parent image cannot set BOARD of the child image when DOMAIN is set")
     endif()
-    set(${ACI_NAME}_BOARD ${ACI_DOMAIN})
-    set(${ACI_DOMAIN}_PM_DOMAIN_DYNAMIC_PARTITION ${ACI_NAME} CACHE STRING "" FORCE)
+
+    if (NOT CONFIG_DOMAIN_${ACI_DOMAIN}_BOARD)
```

Comment:
I think this should be changed to a function arg, and let the caller do the relation between Kconfig argument and the `add_child_image` parameter, to ensure that the API to `add_child_image` is properly described.

Reply:
Check it now @tejlmand